static/js/stream_list.js

```diff
@@ -29,23 +29,51 @@ import * as topic_zoom from "./topic_zoom";
 import * as ui from "./ui";
 import * as ui_util from "./ui_util";
 import * as unread from "./unread";
+import {user_settings} from "./user_settings";
 
 export let stream_cursor;
 
 let has_scrolled = false;
 
+function should_hide_count(sub) {
+    const show_unreads_for = {
+        ALL_STREAMS: user_settings.show_unread_counts === 1,
+        UNMUTED_STREAMS: user_settings.show_unread_counts === 2,
+        NO_STREAMS: user_settings.show_unread_counts === 3,
+    };
+
+    const hide_count =
+        show_unreads_for.NO_STREAMS || (show_unreads_for.UNMUTED_STREAMS && sub.is_muted);
+
+    return hide_count;
+}
```

Comment:
This should I think be done using the coding patterns in the `settings_data` / `settings_config` modules; see `using_dark_theme` for an example of how to do this sort of check nicely. I think a `settings_data.should_mask_unread_count(sub.is_muted)` would be the cleanest API, since we don't want to have to import the types for `sub` objects there and we really just need that boolean to make the determination.